File: leetcode/771-jewels-and-stones/jewels-and-stones.cpp
class Solution {
public:
  int numJewelsInStones(string J, string S) {
    bool jewels[256] = { 0 };
    for (char& ch : J) {
      jewels[ch] = true;
    }

    int result = 0;
    for (char& ch : S) {
      if (jewels[ch]) {
        result++;
      }
    }

    return result;
  }
};

File: leetcode/771-jewels-and-stones/jewels-and-stones.js
function numJewelsInStones(J, S) {
  let result = 0;
  const s = [];
  for (const ch of S) {
    console.log(ch);
    s[ch] = -~s[ch];
  }
  for (const ch of J) {
    result += (s[ch] || 0);
  }
  return result;
};
